src/Polytopes/PolyhedralFan/properties.jl

```diff
@@ -361,3 +361,149 @@ true
+julia> primitive_collections(normal_fan(Oscar.simplex(3)))
+1-element Vector{Vector{Int64}}:
+ [1, 2, 3, 4]
+```
+"""
+function primitive_collections(PF::PolyhedralFan)
+    # collect data
+    cones = [findall(x->x!=0, l) for l in eachrow(pm_fan(PF).MAXIMAL_CONES)]
+    all_points = [i for i in 1 : pm_fan(PF).N_RAYS]
+    d_max = maximum([length(i) for i in cones]) + 1
+    # identify and return the primitive collections
+    collections = Vector{Int}[];
+    for d in 1 : d_max
+        checked  = Vector{Int}[];
+        for cone in cones
+            if d <= length(cone)
+                for I_minus_j in Oscar.Hecke.subsets(cone, d)
+                    scanner = setdiff(all_points, I_minus_j)
+                    for j in scanner
+                        I = vcat(I_minus_j, [j])
+                        if (I in checked) == false
+                            push!(checked, I)
+                            # (1) I is contained in the primitive collections iff it is not contained in any cone
+                            if !(0 in [!issubset(Set(I), Set(test_cone)) for test_cone in cones])
+                                # (2) I is generator of the primitive collections iff primitive_collections does not contain a "smaller" generator
+                                if !(0 in [!issubset(Set(prim), Set(I)) for prim in collections])
+                                    push!(collections, I) # add new generator
+                                end
+                            end
+                        end
+                    end
+                end
+            end
+        end
+    end
+    # return the computed primitive collections
+    return collections;
+end
+
+
+###############################################################################
+## Star subdivision
+###############################################################################
+
+@doc Markdown.doc"""
+    starsubdivision(PF::PolyhedralFan, n::Int)
+
+Computes the star subdivision of a polyhedral fan at its n-th maximal torus orbit.
+
+# Examples
+```jldoctest
+julia> starsubdivision(normal_fan(Oscar.simplex(3)), 1)
+A polyhedral fan in ambient dimension 3
+```
+"""
+function starsubdivision(PF::PolyhedralFan, n::Int)
+    # extract defining information on the fan
+    maxcones = [findall(x->x!=0, l) for l in eachrow(pm_fan(PF).MAXIMAL_CONES)]
+    rays = [[Int(i) for i in r] for r in eachrow(Oscar.Polymake.common.primitive(pm_fan(PF).RAYS))]
+    
+    # check if n-th maximal cone does exist
+    if length(maxcones) < n
+        throw(ArgumentError("Cannot subdivide maximal cone " * string(n) * " as it does not exist!"))
+    end
+    
+    # construct this cone and check if it is smooth
+    gens = zeros(Int64,length(maxcones[n]),length(rays[1]))
+    for i in 1 : length(maxcones[n])
+        for j in 1:length(rays[1])
+            gens[i,j] = rays[maxcones[n][i]][j]
+        end
+    end
+    cone = Oscar.positive_hull(gens)
+    if issmooth(PolyhedralFan(cone)) == false
```

Comment:
A matter of taste, but I find this easier to process mentally:
```suggestion
    if !issmooth(PolyhedralFan(cone))
```
But again, subjective, feel free to ignore

Reply:
I appreciate that change. Thank you @fingolfin .